utils/patterns/pattern.go

```diff
@@ -0,0 +1,698 @@
+		m := map[string]interface{}{}
+		// foundFormatIntOrString := false
+		for k, v2 := range x {
+			if isSchema && k == "enum" { //While schema prettification, ENUMS are end system defined end user input and therefore should not be prettified/deprettified
+				m[k] = v2
+				continue
+			}
+			newmap := ConvertMapInterfaceMapString(v2, prettify, isSchema)
+			if isSchema && isSpecialKey(k) {
+				m[k] = newmap
+			} else if prettify {
+				m[manifests.FormatToReadableString(k)] = newmap
+			} else {
+				m[manifests.DeFormatReadableString(k)] = newmap
+			}
+		}
+		return m
+	case string:
+		if isSchema {
+			if prettify {
+				return manifests.FormatToReadableString(x) //Whitespace formatting should be done at the time of prettification only
+			}
+			return manifests.DeFormatReadableString(x)
+		}
+	}
+	return v
+}
+
+// These keys should not be prettified to "any Of", "all Of" and "one Of"
+var keysToNotPrettifyOnSchema = []string{"anyOf", "allOf", "oneOf"}
+
+func isSpecialKey(k string) bool {
+	for _, k0 := range keysToNotPrettifyOnSchema {
+		if k0 == k {
+			return true
+		}
+	}
+	return false
+}
+
+// In case of any breaking change or bug caused by this, set this to false and the whitespace addition in schema generated/consumed would be removed(will go back to default behavior)
+const Format prettifier = true
+
+type DryRunResponseWrapper struct {
+	//When success is true, error will be nil and Component will contain the structure of the component as it will look after deployment
+	//When success is false, error will contain the errors. And Component will be set to Nil
+	Success   bool            `json:"success"`
+	Error     *DryRunResponse `json:"error"`
+	Component *Service        `json:"component"` //Service is synonymous with Component. Later Service is to be changed to "Component"
+}
+type DryRunResponse struct {
+	Status string
+	Causes []DryRunFailureCause
+}
+
+type DryRunFailureCause struct {
+	Type      string //Type of error
+	Message   string //Error message
+	FieldPath string //Dot separated field path inside service. (For eg: <name>.settings.spec.containers (for pod) or <name>.annotations ) where <name> is the name of service/component
+}
+
+// Pattern is the golang representation of the Pattern
+// config file model
+type Pattern struct {
+	// Name is the human-readable, display-friendly descriptor of the pattern
+	Name string `yaml:"name,omitempty" json:"name,omitempty"`
+	//Vars will be used to configure the pattern when it is imported from other patterns.
+	Vars map[string]interface{} `yaml:"vars,omitempty" json:"vars,omitempty"`
+	// PatternID is the moniker use to uniquely identify any given pattern
+	// Convention: SMP-###-v#.#.#
+	PatternID string              `yaml:"patternID,omitempty" json:"patternID,omitempty"`
+	Services  map[string]*Service `yaml:"services,omitempty" json:"services,omitempty"`
+}
+
+// Service represents the services defined within the appfile
+type Service struct {
```

Comment:
Outside of backwards compatibility and other changes that would be needed, there is no longer a need for `Service`, is there?

Reply:
> there is no longer a need for Service

Yes.
The pattern engine will also require changes.